server/repository/articleRepository.js

```diff
@@ -57,21 +50,43 @@ class ArticleRepository {
   }
 
   /**
-   * Function to get a single article
-   * @param { string } slug
-   * @returns { object }
-   ** otherwise it throws an error
-   */
-  static async getArticleBySlug(slug) {
-    const singleArticle = await Articles.findOne({
-      where: {
-        slug
-      }
+  * Finds an article by parameter supplied
+  * @param {string} param Parameter to search by
+  * @param {string} value Parameter value
+  * @returns {object | null} Article object or null if article is not found
+  */
+  static async getArticleByParam(param, value) {
```

Comment:
can this function take multiple params as opposed to one?  I prefer the  getArticleBySlug method because its simple and straight to the point.

Reply:
@seyi-adeleke, Victor worked on this. We thought of a situation similar to the user where we might want to get an article by different parameters. That is the reason for this generic function.